ada/src/stemmer-factory.adb

```diff
@@ -0,0 +1,271 @@
+-----------------------------------------------------------------------
+--  stemmer-factory -- Multi-language stemmer with Snowball generator
+--  Copyright (C) 2020 Stephane Carrez
+--  Written by Stephane Carrez (Stephane.Carrez@gmail.com)
+--
+--  Licensed under the Apache License, Version 2.0 (the "License");
+--  you may not use this file except in compliance with the License.
+--  You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+--  Unless required by applicable law or agreed to in writing, software
+--  distributed under the License is distributed on an "AS IS" BASIS,
+--  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+--  See the License for the specific language governing permissions and
+--  limitations under the License.
+-----------------------------------------------------------------------
+with Stemmer.Arabic;
+with Stemmer.Basque;
+with Stemmer.Catalan;
+with Stemmer.Danish;
+with Stemmer.Dutch;
+with Stemmer.English;
+with Stemmer.Finnish;
+with Stemmer.French;
+with Stemmer.German;
+with Stemmer.Greek;
+with Stemmer.Hindi;
+with Stemmer.Hungarian;
+with Stemmer.Indonesian;
+with Stemmer.Irish;
+with Stemmer.Italian;
+with Stemmer.Lithuanian;
+with Stemmer.Nepali;
+with Stemmer.Norwegian;
+with Stemmer.Porter;
+with Stemmer.Portuguese;
+with Stemmer.Romanian;
+with Stemmer.Russian;
+with Stemmer.Serbian;
+with Stemmer.Spanish;
+with Stemmer.Swedish;
+with Stemmer.Tamil;
+with Stemmer.Turkish;
```

Comment:
We want to avoid hard-coded lists of languages that have to be manually updated for each new algorithm, because that adds a lot of tedious work for adding a new algorithm, and mechanical updates done by hand tend to miss places that need updating or get copied and pasted from an existing case but not fully updated.  So all the other backends generate any code like this that they need at build time, so currently the only update needed is to add the new algorithm to `libstemmer/modules.txt`.

We can probably just use essentially the same approach as for pascal (`pascal/generate.pl` working from `stemwords-template.dpr` which has marked blocks of code that get expanded for each stemming algorithm with the name inserted.  

Unless there's some Ada-specific way you'd rather use (e.g. Rust does this in `rust/build.rs`), I can probably hook this up based on what's in the PR (I don't know any Ada, but I don't think I really need to).